yabt/utils.py

```diff
@@ -101,10 +101,8 @@ def link_node(abs_src: str, abs_dest: str, force: bool=False):
         # sync file by linking it to dest
         link_func(abs_src, abs_dest, force)
     elif isdir(abs_src):
-        # sync dir by recursively linking files under it to dest
-        shutil.copytree(abs_src, abs_dest,
-                        copy_function=link_func,
-                        ignore=shutil.ignore_patterns('.git'))
+        raise ValueError('try to link: {} which is a directory and not a file',
```

Comment:
@itamaro I couldn't find when we are using this with directories, so I replaced it with raise ValueError
Do you know any reason why this should be called with a directory?

Reply:
this function can be given user-provided paths through YBuilds, which can generally also be directories (don't know if it's in active use or not in existing projects)

Reply:
When building search-stack this code is never reached.
Does this mean we can remove this? is there any other place that can use it?

Reply:
maybe when rebuilding base images?

I also suspect it would be used by some frontend frameworks to copy static resources and such, and maybe also ML stuff to copy pretrained data files

Reply:
do you have a specific reason you want to drop this flow?

Reply:
Yes. There is a bug in this flow - the flag `force` is not passed to link_func as it is passed in the other flow.
I wanted to fix it but I couldn't get this code to run, so I can't know if I fixed it correctly.
The fix I wanted is something like:
`copy_function=functions.partial(link_func, force=force)`

Reply:
We found what makes it run (FileGroup for example)
So I removed the ValueError, fixed the code and added a test that covers this place.